### PR TITLE
fix user associated with expenses

### PR DESCRIPTION
**Pull request by asood123**
Currently, when a new expense is created, we set the userid in transactions table to be the host. This fixes it back to the person who filed the expense.

### CODE REVIEW

File: test/expenses.routes.test.js [hunk 1]
```diff
@@ -829,8 +831,7 @@ describe('expenses.routes.test.js', () => {
     let group, user;
     return (g ? Promise.resolve(g) : models.Group.create(utils.data('group2')))
       .tap(g => group = g)
-      .then(() => u ? u : models.User.create(utils.data('user4'))
-        .tap(user => group.addUserWithRole(user, roles.HOST)))
+      .then(() => u ? u : expenseFiler)
```
Comment (xdamman) [inline test/expenses.routes.test.js]: Not a fan of this function. `createExpense` should not be responsible to create a group. Also, this particular line is very hard to follow. Why is a user turning into `expenseFiler`?

**Comment by asood123** [comment]
I don't think it's worth reworking the tests for this small change. Took me
over 2 hours to make these small changes to the tests.
